reactComponents/src/searchbar/redux/tokenSearchSlice.ts

```diff
@@ -25,91 +26,56 @@ export const resetSearchOnNewExchange = createAsyncThunk(
-  async (searchString:any, thunkAPI:any) => {
+  async (searchString, thunkAPI) => {
+    // Cancel search is string is empty.
+    if (!searchString) return;
+
     try {
       let { networkMap, exchangeMap } = thunkAPI.getState();
-      let processedNetworks;
-      let processedExchanges;
       const pairSearchTimestamp = new Date().getTime();
+      console.log('Search sent');
 
 
       // Dispatches "setPairSearchTimestamp".
       thunkAPI.dispatch(setPairSearchTimestamp(pairSearchTimestamp));
 
       // Runs the function handling the cleaning of the properties from their values indicating if they are enabled or not.
-      [processedNetworks, processedExchanges] = valueCleaner(networkMap, exchangeMap);
+      networkMap = filterActiveNames(networkMap);
+      exchangeMap = filterActiveNames(exchangeMap);
 
       // Runs the function handling the management of the "All" value selected by the user.
-      [processedNetworks, processedExchanges] = allValueHandler(processedNetworks, processedExchanges);
+      networkMap = filterActiveAll(networkMap) ? [...new Set(networkExchangePairs.map(pair => pair[0]))] : networkMap;
+      exchangeMap = filterActiveAll(exchangeMap) ? [...new Set(networkExchangePairs.map(pair => pair[1]))] : exchangeMap;
 
       // Filtering out any exchange that is not valid for the selected networks.
       // This has to be done since an exchange will remain in the array when the network is disabled by the user.
       // It's easier here and also offer a more natural experience for the user.
-      processedExchanges = processedExchanges
+      exchangeMap = exchangeMap
         .filter(exchange => networkExchangePairs
-          .filter(pair => processedNetworks.includes(pair[0]) && pair[1] === exchange).length >= 1);
+          .filter(pair => networkMap.includes(pair[0]) && pair[1] === exchange).length >= 1);
 
       // Filtering out any network that does not have at least one valid exchange selected.
       // This has to be done since the user can still have a network selected while it has no valid exchange selected.
       // It's easier here and also offer a more natural experience for the user.
       // We do this in last because this has a real potential to harm the user experience by running GraphQL queries that are not needed unlike feeding to the
       // query an unused echange for a given network.
-      processedNetworks = processedNetworks
+      networkMap = networkMap
         .filter(network => networkExchangePairs
-          .filter(pair => pair[0] === network && processedExchanges.includes(pair[1])).length >= 1);
+          .filter(pair => pair[0] === network && exchangeMap.includes(pair[1])).length >= 1);
 
       // Loading the data.
-      const data = await retry(() => searchTokensAsync(searchString, processedNetworks, processedExchanges), { retries: 1 });
+      const data = await retry(() => searchTokensAsync(searchString, networkMap, exchangeMap), { retries: 1 });
```

Comment:
why do we need to retry?

Reply:
We need it, it's to get the token pair by network and exchange, although it's not what I wrote.